core/src/main/java/com/linecorp/armeria/common/RequestContextAwareEventLoop.java

```diff
@@ -48,7 +48,7 @@ protected EventLoop delegate() {
 
     @Override
     public EventLoop next() {
-        return delegate().next();
+        return new RequestContextAwareEventLoop(context(), delegate().next());
```

Comment:
I think in practice, this will just generate garbage for no reason when the event loop is a single thread, which IIUC is always the case.

How about we handle it like this

```java
EventLoop next = delegate().next();
return next == delegate() ? this : new RequestContextAwareEventLoop(context, next);
```

@trustin Please confirm as I'm only guessing on netty's behavior :)